src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -1905,7 +1957,13 @@ class MutateSomePartColumnsTask : public IExecutableTask
 
         (*ctx->mutate_entry)->columns_written = ctx->storage_columns.size() - ctx->updated_header.columns();
 
-        ctx->new_data_part->checksums = ctx->source_part->checksums;
```

Comment:
This is the bug prone line. I remove it.
Now we construct `new_data_part->checksums` more selectively from `ctx->source_part->checksums` by picking particular files.